redisson/src/main/java/org/redisson/cluster/ClusterConnectionManager.java

```diff
@@ -502,7 +502,7 @@ private ClusterPartition find(Collection<ClusterPartition> partitions, Integer s
         for (final ClusterPartition newPart : newPartitions) {
             boolean masterFound = false;
             for (ClusterPartition currentPart : getLastPartitions()) {
-                if (!newPart.getMasterAddress().equals(currentPart.getMasterAddress())) {
+                if (null != newPart && null != newPart.getMasterAddress() && !newPart.getMasterAddress().equals(currentPart.getMasterAddress())) {
```

Comment:
imo this is sweeping the problem under the carpet, the fact newpart or master address is null seems like a bug / validation should be done earlier.